mediator/esmFldsExchange_cesm_mod.F90

```diff
@@ -2519,6 +2548,263 @@ subroutine esmFldsExchange_cesm(gcomp, phase, rc)
+      else
+         if ( fldchk(is_local%wrap%FBExp(compocn)         , 'Sw_ustokes_wavenumber_6', rc=rc) .and. &
+              fldchk(is_local%wrap%FBImp(compwav, compwav), 'Sw_ustokes_wavenumber_6', rc=rc)) then
+            call addmap_from(compwav, 'Sw_ustokes_wavenumber_6', compocn,  mapbilnr_nstod, 'one', wav2ocn_map)
+            call addmrg_to(compocn, 'Sw_ustokes_wavenumber_6', mrg_from=compwav, mrg_fld='Sw_ustokes_wavenumber_6', mrg_type='copy')
+         end if
+      end if
+    end if
+    !-----------------------------
+    ! to ocn: 
+    !-----------------------------
+    if (ocn_name == 'mpaso') then
+      if (phase == 'advertise') then
+         call addfld_from(compwav, 'Sw_vstokes_wavenumber_6')
+         call addfld_to(compocn, 'Sw_vstokes_wavenumber_6')
+      else
+         if ( fldchk(is_local%wrap%FBExp(compocn)         , 'Sw_vstokes_wavenumber_6', rc=rc) .and. &
+              fldchk(is_local%wrap%FBImp(compwav, compwav), 'Sw_vstokes_wavenumber_6', rc=rc)) then
+            call addmap_from(compwav, 'Sw_vstokes_wavenumber_6', compocn,  mapbilnr_nstod, 'one', wav2ocn_map)
+            call addmrg_to(compocn, 'Sw_vstokes_wavenumber_6', mrg_from=compwav, mrg_fld='Sw_lamult', mrg_type='copy')
+         end if
+      end if
+    end if
+    !-----------------------------
+    ! to ocn: 
+    !-----------------------------
+    if (ocn_name == 'mpaso') then
+      if (phase == 'advertise') then
+         call addfld_from(compwav, 'Sw_ustokes_wavenumber_6')
+         call addfld_to(compocn, 'Sw_ustokes_wavenumber_6')
+      else
+         if ( fldchk(is_local%wrap%FBExp(compocn)         , 'Sw_ustokes_wavenumber_6', rc=rc) .and. &
+              fldchk(is_local%wrap%FBImp(compwav, compwav), 'Sw_ustokes_wavenumber_6', rc=rc)) then
+            call addmap_from(compwav, 'Sw_ustokes_wavenumber_6', compocn,  mapbilnr_nstod, 'one', wav2ocn_map)
+            call addmrg_to(compocn, 'Sw_ustokes_wavenumber_6', mrg_from=compwav, mrg_fld='Sw_ustokes_wavenumber_6', mrg_type='copy')
+         end if
+      end if
+    end if
+    !-----------------------------
+    ! to ocn: 
+    !-----------------------------
+    if (ocn_name == 'mpaso') then
+      if (phase == 'advertise') then
+         call addfld_from(compwav, 'Sw_vstokes_wavenumber_6')
+         call addfld_to(compocn, 'Sw_vstokes_wavenumber_6')
+      else
+         if ( fldchk(is_local%wrap%FBExp(compocn)         , 'Sw_vstokes_wavenumber_6', rc=rc) .and. &
+              fldchk(is_local%wrap%FBImp(compwav, compwav), 'Sw_vstokes_wavenumber_6', rc=rc)) then
+            call addmap_from(compwav, 'Sw_vstokes_wavenumber_6', compocn,  mapbilnr_nstod, 'one', wav2ocn_map)
+            call addmrg_to(compocn, 'Sw_vstokes_wavenumber_6', mrg_from=compwav, mrg_fld='Sw_lamult', mrg_type='copy')
+         end if
+      end if
+    end if
+
+
+    !-----------------------------
+    ! to ocn:
+    !-----------------------------
+    if (ocn_name == 'mpaso') then
```

Comment:
I think that you can bracket all of this code with one if instead of having one for each variable.

Reply:
Understood, will do.